### PR TITLE
build(gradle): javadoc enhance + namings migration

### PR DESCRIPTION
**Pull request by stachu540**
### Prerequisites for Code Changes

* [x] This pull request follows the code style of the project
* [x] I have tested this feature

### Changes Proposed

* Package Groups - see [this example](https://github.com/lenskit/lenskit/blob/master/gradle/javadoc.gradle#L64-L92)
* `aggregateJavadoc` task behavior changes
  * removes directory while activating task
  * clones `element-list` as `package-list` in the same directory
  * task is active if Gradle using JVM9+
* adding some links which points to specific class documentation. (some of them doesn't have a `package-list` file attached into index of javadoc)
* small fix javadoc when it should not be a tag which recognize as unknown while compile javadoc. Adds code block in this fix.
* adding overview files
* Javadoc generates into `en` localization - no matter which localization is on your PC
* obsolete unnecessary `artifactId` from `buildSrc`
* migrate module names into `settings.gradle.kts`

### Additional Information

This is proposal changes to javadoc, which adds Groups into specific packages, changes to javadoc aggregation and adding some links to the other javadoc documentation. Got adding some improvments with Javadoc Overview, constant localization and module names migrated to `settings.gradle.kts`. Each changes are tested. Only left is some typings, some improvments and it is ready to review.

### CODE REVIEW

**Comment by PhilippHeuer** [comment]
i get the following 2 errors when running aggregateJavadoc, javadoc works fine. It seems like aggregateJavadoc is more strict on validation.

```
...\twitch4j\buildSrc\overview-general.html:59: error: no summary or caption for table
</table>
^
...\twitch4j\buildSrc\overview-general.html:60: error: heading used out of sequence: <H3>, compared to previous heading: <H1>
<h3>To getting started please refer to our documentation</h3>
```

I think we need to replace the h3 with h2 in both html templates and add a <caption>Heading</caption> element inside of the `package name / group name` table.

**Comment by stachu540** [comment]
got made some tweaks in requested files

**Review by PhilippHeuer — APPROVED** [review]
I made a few more changes - the name of one module was wrong and i simplified the templates a little because they took up to much space - i think a few useful links are enough, most information should be in the documentation.

File: buildSrc/overview-general.html [hunk 1]
```diff
@@ -1,79 +1,11 @@
...
-		<td>Twitch Message Interface - API</td>
-	</tr>
-	<tr>
-		<td style="float: right"><code>twitch4j</code></td>
-		<td>Core</td>
-	</tr>
-</table>
-<h2>To getting started please refer to our documentation</h2>
+<p><img alt="Twitch4J Logo" width="64" height="64" src="https://twitch4j.github.io/assets/images/logo.svg"/></p>
+<p>Check out our documentation at <a href="https://twitch4j.github.io/">twitch4j.github.io</a></p>
```
Comment (stachu540) [inline buildSrc/overview-general.html:4]:
```suggestion
<p>Check out our documentation at <a href="https://twitch4j.github.io/docs">twitch4j.github.io</a></p>
```

File: buildSrc/overview-general.html [hunk 2]
```diff
@@ -1,79 +1,11 @@
...
-<h2>Thank you to our sponsors to provide open-source developing tools</h2>
-<ul>
-	<li>
-		<a href="https://www.jetbrains.com/">JetBrains</a> - for providing <a
-			href="https://www.jetbrains.com/idea/">IntelliJ IDEA</a></li>
-	<li>
-		<a href="https://www.yourkit.com/">YourKit</a> - for providing <a
-			href="https://www.yourkit.com/java/profiler/">Java Profiler</a> for Performance Tuning
-	</li>
+	<li><a href="https://twitch4j.github.io/javadoc">Aggregated Javadocs (all modules)</a></li>
```
Comment (stachu540) [inline buildSrc/overview-general.html:6]: It should be in `overview-single.html`. If someone gonna checks specific module in Javadoc.io it needs redirect to the main one.

File: buildSrc/overview-single.html [hunk 3]
```diff
@@ -1,30 +1,10 @@
...
-	<a href="https://twitch4j.github.io">
-		<img alt="Twitch4J Logo" width="128" height="128" src="https://twitch4j.github.io/assets/images/logo.svg"/>
-	</a>
-</p>
-<h1 style="text-align: center">Welcome to the Twitch4J Javadoc Documentation</h1>
-<p style="text-align: center">This javadoc is a partial of the full documentation hosted into <a href="https://twitch4j.github.io/javadoc">official
-	website</a></p>
-<h2>To getting started please refer to our documentation</h2>
+<p><img alt="Twitch4J Logo" width="64" height="64" src="https://twitch4j.github.io/assets/images/logo.svg"/></p>
+<p>Check out our documentation at <a href="https://twitch4j.github.io/">twitch4j.github.io</a></p>
```
Comment (stachu540) [inline buildSrc/overview-single.html:4]:
```suggestion
<p>Check out our documentation at <a href="https://twitch4j.github.io/docs">twitch4j.github.io</a></p>
```

**Review by stachu540 — COMMENTED** [review]
One last take before we getting into it.

**Comment by stachu540** [comment]
LGTM

Looks like we can push this one.

**Pull request merged**